torch/csrc/jit/codegen/cuda/scheduler/mma_utils.cpp

```diff
@@ -171,12 +171,30 @@ namespace {
 //  to the given mma dimension. See [MMA dimension matching].
 std::vector<IterDomain*> getMmaDomains(MmaOp* mma, MmaDimension dimension) {
   // This utility is user facing so shouldn't ever see tensor index here.
-  auto accumulator_domain =
-      mma->out()->as<TensorView>()->getMaybeRFactorDomain();
-  auto a_domain = TensorDomain::noReductions(
-      mma->inA()->as<TensorView>()->getMaybeRFactorDomain());
-  auto b_domain = TensorDomain::noReductions(
-      mma->inB()->as<TensorView>()->getMaybeRFactorDomain());
+
+  // Note:
+  //  Have to use root domain here since the operands do not have
+  //  root/rfactor domains that map to the rfactor domain of output.
+  //  For example:
+  //   C[I,I,R,R] = mma (A[I,B,I,I], B[B,I,I,I]),
+  //  if we do
+  //    c->split(-1,4);
+  //    c->rfactor(-1);
+  //  on the mma stage we get:
+  //   C[I,I,R,Io,R(4)] = mma (A[I,B,I,I], B[B,I,I,I]),
+  //  and in this case Io and R(4) would not be able to find root mapping
+  //  in A or B.
+  //
+  //  Essentially in the case of rfactor, this utility does producer side
+  //   matching so looking at root domain would be required.
+  //  This matching pattern should support most common matmul applications,
+  //   but in follow ups we may need to extend RFactor matching if there
+  //   are more complex scheduling patterns that we want to support.
+  auto accumulator_domain = mma->out()->as<TensorView>()->getRootDomain();
```

Comment:
I can understand out being root domain, but why should `a` and `b` be root instead of `rfactor`?
Wouldn't this be an issue if `a` or `b` was an output of a previous fusion and a result of a reduction?

Reply:
I guess similar concern of if a view immediately precedes a matmul.